skrifa/src/outline/glyf/hint/engine/data.rs

```diff
@@ -1,12 +1,108 @@
 //! Reading and writing data.
 //!
-//! Implements 2 instructions.
+//! Implements 7 instructions.
 //!
 //! See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#reading-and-writing-data>
 
-use super::{Engine, OpResult};
+use super::{
+    super::{graphics_state::ZonePointer, math},
+    Engine, OpResult,
+};
 
 impl<'a> Engine<'a> {
+    /// Get coordinate project in to the projection vector.
+    ///
+    /// GC\[a\] (0x46 - 0x47)
+    ///
+    /// a: 0: use current position of point p
+    ///    1: use the position of point p in the original outline
+    ///
+    /// Pops: p: point number
+    /// Pushes: value: coordinate location (F26Dot6)
+    ///
+    /// Measures the coordinate value of point p on the current
+    /// projection_vector and pushes the value onto the stack.
+    ///
+    /// See <https://learn.microsoft.com/en-us/typography/opentype/spec/tt_instructions#get-coordinate-projected-onto-the-projection_vector>
+    /// and <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttinterp.c#L4512>
+    pub(super) fn op_gc(&mut self, opcode: u8) -> OpResult {
+        let p = self.value_stack.pop_usize()?;
+        let value = if (opcode & 1) != 0 {
+            self.graphics_state
+                .dual_project(self.graphics_state.zp2().original(p)?, Default::default())
+        } else {
+            self.graphics_state
+                .project(self.graphics_state.zp2().point(p)?, Default::default())
+        };
```

Comment:
Optional, perhaps easier to read if you bind `self.graphics_state.project(self.graphics_state.zp2()` so the if branches are only whats different

Reply:
One uses `dual_project` on the original point and the other uses `project` on the scaled point so there isn't much commonality but I bound `self.graphics_state` to a local like the other functions and this cleaned up some of the noise.